File: api/python/test/clip.py
#!/usr/bin/env python

import sys
import numpy
import m8r

par = m8r.Par()
inp  = m8r.Input()
output = m8r.Output()
assert 'float' == inp.type

n1 = inp.int("n1")
n2 = inp.size(1)
assert n1

clip = par.float("clip")
assert clip

trace = numpy.zeros(n1,'f')

for i2 in range(n2): # loop over traces
    inp.read(trace)
    trace = numpy.clip(trace,-clip,clip)
    output.write(trace)

sys.exit(0)

File: api/julia/test/clip.jl
#!/usr/bin/env julia

using m8r

inp = m8r.input("in")
out = m8r.output("out")

n1 = m8r.histint(inp, "n1")
n2 = m8r.leftsize(inp, 1)

clip = m8r.getfloat("clip")

trace = Array{Float32}(undef, n1)

for i2 in 1:n2
    m8r.floatread(trace, n1, inp)
    clamp!(trace, -clip, clip)
    m8r.floatwrite(trace, n1, out)
end

